### PR TITLE
Added base_name parameter to Density material class. Made all module base_name patterns constant.

### PR DESCRIPTION
**Pull request by jbtompkins**
refs #13600 
Added the base_name optional parameter to the Density material class to be able to use Density in BiMaterial simulations.

### CODE REVIEW

**Comment by jbtompkins** [comment]
Requesting @bwspenc review this PR after tests complete.

File: modules/misc/include/materials/Density.h [hunk 1]
```diff
@@ -28,6 +28,7 @@ class Density : public Material
   std::vector<const VariableGradient *> _grad_disp;
   const VariableValue & _disp_r;
 
+  std::string _base_name;
```
Comment (dschwen) [inline modules/misc/include/materials/Density.h]:
```suggestion
  const std::string _base_name;
```

(I'll make a PR to change the rest to const as well)

**Review by dschwen — DISMISSED** [review]
This is pretty uncontroversial. I made one suggestion, but it should not necessarily hold up a merge.

File: modules/misc/include/materials/Density.h (hunk 1 above)
Comment (jbtompkins) [reply modules/misc/include/materials/Density.h]: I agree with adding const and have verified that it does not break anything. Added suggested commit to this PR.

What did you mean by your comment "I'll make a PR to change the rest to const as well?"

Comment (dschwen) [reply modules/misc/include/materials/Density.h]: > What did you mean by your comment "I'll make a PR to change the rest to const as well?"

All classes that use the base name pattern should be consistent. Right now some have this member marked const and some don't. This should be fixed.

**Review by dschwen — DISMISSED** [review]

File: modules/misc/include/materials/Density.h (hunk 1 above)
Comment (jbtompkins) [reply modules/misc/include/materials/Density.h]: Right, okay. I can fix those and include them in this PR.

**Comment by jbtompkins** [comment]
@dschwen I made all of the instances of _base_name const for any classes that contain that variable. Please let me know if this is what you had in mind.

**Review by permcody — APPROVED** [review]

**Pull request merged**